src/sentry/runner/commands/cleanup.py

```diff
@@ -147,77 +146,76 @@ def is_filtered(model):
                     project=project or '*',
                 )
             )
+
+        # TODO: filtering won't take into account "app-level" cascades done in deletions code...
+        #       e.g. filtering out GroupTagValue won't work when we delete a Group here, because all
+        #            child relations are handled within the deletions code.
+        # TODO: implement concurrency like BulkDeleteQuery has (?)
         if is_filtered(model):
             if not silent:
                 click.echo('>> Skipping %s' % model.__name__)
         else:
-            BulkDeleteQuery(
+            query = {
+                '{}__lte'.format(dtfield): (timezone.now() - timedelta(days=days)),
+            }
+
+            if project_id:
+                if 'project' in model._meta.get_all_field_names():
+                    query['project'] = project_id
+                else:
+                    query['project_id'] = project_id
+
+            task = deletions.get(
                 model=model,
-                dtfield=dtfield,
-                days=days,
-                project_id=project_id,
-                order_by=order_by,
-            ).execute()
-    # EventMapping is fairly expensive and is special cased as it's likely you
-    # won't need a reference to an event for nearly as long
-    if not silent:
-        click.echo("Removing expired values for EventMapping")
-    if is_filtered(models.EventMapping):
-        if not silent:
-            click.echo('>> Skipping EventMapping')
-    else:
-        BulkDeleteQuery(
-            model=models.EventMapping,
-            dtfield='date_added',
-            days=min(days, 7),
-            project_id=project_id,
-            order_by='-date_added'
-        ).execute()
+                query=query,
+                transaction_id=uuid4().hex,
+            )
 
-    # Clean up FileBlob instances which are no longer used and aren't super
-    # recent (as there could be a race between blob creation and reference)
-    if not silent:
-        click.echo("Cleaning up unused FileBlob references")
-    if is_filtered(models.FileBlob):
-        if not silent:
-            click.echo('>> Skipping FileBlob')
-    else:
-        cleanup_unused_files(silent)
+            has_more = True
+            while has_more:
+                has_more = task.chunk()
+
+    # Deletions that use `BulkDeleteQuery` (and don't need to worry about child relations)
+    # (model, datetime_field, max_days)
+    BULK_QUERY_DELETES = (
+        (models.TagValue, 'last_seen', None),
+        (models.EventMapping, 'date_added', 7),
+    )
```

Comment:
You might be asking yourself, "why aren't these just in the tuple above, to be handled by the deletions code path?"

The problem is that they both use `BulkModelDeletionTask` in the deletions code:

https://github.com/getsentry/sentry/blob/8330a0938a896d813a342be1e1ca2237da4a6074/src/sentry/deletions/__init__.py#L73

https://github.com/getsentry/sentry/blob/8330a0938a896d813a342be1e1ca2237da4a6074/src/sentry/deletions/__init__.py#L45

`ModelDeletionTask` passes the `query` down to the Django ORM:

https://github.com/getsentry/sentry/blob/8330a0938a896d813a342be1e1ca2237da4a6074/src/sentry/deletions/base.py#L134

But `BulkModelDeletionTask` inherits from `ModelDeletionTask` but then uses the `query` in a hardcoded `WHERE x = y`:

https://github.com/getsentry/sentry/blob/96bc1c63df5ec73fe12c136ada11561bf52f1ec9/src/sentry/utils/query.py#L292-L293

In short, it doesn't currently support doing `date_field < value`. Since they don't have child relations anyway, I've left them on the (faster) `BulkDeleteQuery` path for now.